src/hb-ot-shape-fallback.cc

```diff
@@ -525,6 +525,13 @@ _hb_ot_shape_fallback_spaces (const hb_ot_shape_plan_t *plan HB_UNUSED,
   for (unsigned int i = 0; i < count; i++)
     if (_hb_glyph_info_is_unicode_space (&info[i]) && !_hb_glyph_info_ligated (&info[i]))
     {
+      /* If font had no ASCII space and we used the invisible glyph, give it a 1/4 EM default advance. */
+      if (buffer->invisible && info[i].codepoint == buffer->invisible)
```

Comment:
This gets a warning here:

../src/hb-ot-shape-fallback.cc:529:10: warning: suggest explicit braces to avoid ambiguous ‘else’ [-Wdangling-else]
  529 |       if (buffer->invisible && info[i].codepoint == buffer->invisible)